elb/elb.go

```diff
@@ -190,19 +211,31 @@ func FindFrontEndElbs(awsElb ELB, labelValue string) (map[string]LoadBalancerDet
 		}
 
 		// todo cb error out if we already have an internal or public facing elb
-		for _, description := range output.TagDescriptions {
-			for _, tag := range description.Tags {
-				if *tag.Key == ElbTag && *tag.Value == labelValue {
-					log.Infof("Found frontend elb %s", *description.LoadBalancerName)
-					lb := allLbs[*description.LoadBalancerName]
-					clusterFrontEnds[lb.Scheme] = lb
-				}
+		for _, elbDescription := range output.TagDescriptions {
+			if tagsDoMatch(elbDescription.Tags, requiredTags) {
+				log.Infof("Found frontend elb %s", *elbDescription.LoadBalancerName)
+				lb := allLbs[*elbDescription.LoadBalancerName]
+				clusterFrontEnds[lb.Scheme] = lb
 			}
 		}
 	}
 	return clusterFrontEnds, nil
 }
 
+func tagsDoMatch(elbTags []*aws_elb.Tag, tagsToMatch map[string]string) bool {
+	matches := 0
+	for name, value := range tagsToMatch {
+		log.Debugf("Checking for %s tag set to %s", name, value)
+		for _, elb := range elbTags {
+			if name == *elb.Key && value == *elb.Value {
+				matches++
+			}
+		}
+	}
+
+	return matches == len(tagsToMatch)
+}
```

Comment:
I think we should approach this differently, so that we don't loop over the elbTags for each tag to Match:
1. convert the elbTags to a map first (1 loop), 
2. iterate over the tagsToMatch (1 loop) 
   1. if the tag to match key and value are not on the elb tags map, return false.
3. return true.